tests/btrfs-progs/generate_report.pm

```diff
@@ -29,23 +28,38 @@ sub log_end {
     assert_script_run($cmd);
 }
 
+# Compress all sub directories under $dir and upload them.
+sub upload_subdirs {
+    my ($dir, $timeout) = @_;
+    my $output = script_output("if [ -d $dir ]; then find $dir -maxdepth 1 -mindepth 1 -type f -or -type d; else echo $dir folder not exist; fi");
+    if ($output =~ /folder not exist/) { return; }
+    for my $subdir (split(/\n/, $output)) {
+        my $tarball = "$subdir.tar.xz";
+        assert_script_run("tar cJf $tarball -C $dir " . basename($subdir), $timeout);
+        upload_logs($tarball, timeout => $timeout, log_name => basename($dir));
+    }
+}
+
 sub run {
     my $self = shift;
     $self->select_serial_terminal;
 
     # Finalize status log and upload it
-    log_end($STATUS_LOG);
-    upload_logs($STATUS_LOG, timeout => 60, log_name => "test");
+    log_end STATUS_LOG;
+    upload_logs(STATUS_LOG, timeout => 60, log_name => "test");
+
+    # Upload test logs
+    upload_subdirs(LOG_DIR, 600);
 
     #upload system log
     upload_system_logs();
 
     # Junit xml report
-    my $script_output = script_output("cat $STATUS_LOG", 600);
+    my $script_output = script_output('cat ' . STATUS_LOG, 600);
     my $tc_result     = analyzeResult($script_output);
     my $xml           = generateXML($tc_result);
-    assert_script_run("echo \'$xml\' > $JUNIT_FILE", 7200);
-    parse_junit_log($JUNIT_FILE);
+    assert_script_run("echo \'$xml\' > " . JUNIT_FILE, 7200);
```

Comment:
7200 is 2 hours , this would make sense only of your tests using MAX_JOB_TIME variable with bigger value . otherwise this timeout would be unachievable because in 2 hours scheduler will unconditionally kill the job anyway